### PR TITLE
[READY] Port Fel's fermentation barrel from tg

### PR DESCRIPTION
**Pull request by variableundefined**
**What does this PR do:**
I just want to put this PR up before I go to sleep - It is almost complete, I didn't incude the dmi file and the changes to hydroponics, but it compile.

It ports in Fel / Fludd12's fermentation barrel from tgstation/tgstation#38977 which allow you to brew your own fruit wine with fruits!

Noticeably I didn't include tastes inheritance for fruit wine so a lot of alcohol might end up a bit tasteless - This is working as intended as I do not want to rip out our current taste system for this PR and would like to leave it for another PR instead. 

Changes: 
1. Add in wine power & other variable to other thing
2. Ash flora snacks typepath changed so you can use them in fermentation barrel
3. Fermentation barrel added. They're damageable.
4. Minor related refactor for reagent containers so they can be used to fill a fermentation barrel.

Note: 
Please make sure I converted wine power properly to a percentage system so people don't end up ingesting 3000 proof fruit wine by accident.

Fixes: https://github.com/ParadiseSS13/Paradise/pull/9514
https://github.com/ParadiseSS13/Paradise/pull/9520
https://github.com/ParadiseSS13/Paradise/pull/9554

**Changelog:**
:cl: Fel
add: You can now craft wooden barrels with 30 pieces of wood! They can hold up to 300u of reagents.
add: Putting hydroponics-grown plants into the barrel lets you ferment them for alcohol! Some give bar drinks, while others simply give plain fruit wine.
tweak: Reagent containers like bottle & glass should be more consistent now.
/:cl:

### CODE REVIEW

**Comment by fludd12** [comment]
:smirk_cat: :point_right: :point_right: 

Tell me if you need any help with stuff!

File: code/modules/hydroponics/grown/ambrosia.dm [hunk 1]
```diff
@@ -74,6 +76,8 @@
 	origin_tech = "biotech=6;materials=5"
 	light_range = 3
 	seed = /obj/item/seeds/ambrosia/gaia
+	wine_power = 0.7
+	wine_flavor = "the earthmothert's blessing"
```
Comment (alex-gh) [inline code/modules/hydroponics/grown/ambrosia.dm]: Typo, or some reference I'm missing?

Comment (variableundefined) [reply code/modules/hydroponics/grown/ambrosia.dm]: Definitely a typo, woops

File: code/modules/hydroponics/grown/beans.dm [hunk 2]
```diff
@@ -25,6 +25,7 @@
 	icon_state = "soybeans"
 	filling_color = "#F0E68C"
 	bitesize_mod = 2
+	wine_power = 20
```
Comment (alex-gh) [inline code/modules/hydroponics/grown/beans.dm]: If wine_power is only supposed to go to 1.2 is 20 a bit much?

File: code/modules/hydroponics/grown/beans.dm [hunk 3]
```diff
@@ -45,4 +46,5 @@
 	desc = "Something about these seems fishy."
 	icon_state = "koibeans"
 	filling_color = "#F0E68C"
-	bitesize_mod = 2
\ No newline at end of file
+	bitesize_mod = 2
+	wine_power = 40
```
Comment (alex-gh) [inline code/modules/hydroponics/grown/beans.dm]: Maybe too high.

File: code/modules/hydroponics/fermenting_barrel.dm [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+/obj/structure/fermenting_barrel/attackby(obj/item/I, mob/user, params)
+	var/obj/item/reagent_containers/food/snacks/grown/G = I
+	if(istype(G))
+		if(!G.can_distill)
+			to_chat(user, "<span class='warning'>You can't distill this into anything...</span>")
+			return FALSE
+		else if(!user.drop_item())
+			to_chat(user, "<span class='warning'>[G] is stuck to your hand!</span>")
+			return FALSE
+		G.loc = src
```
Comment (tigercat2000) [inline code/modules/hydroponics/fermenting_barrel.dm]: G.forceMove(src)

**Pull request merged**